Attempting to fix workflows and unit tests

HeroesDieYoung opened the pull request:
- Added pytest-isolate to fix test issues interfering with each other
- Remove codium test files entirely
- Commented out 7 remaining failing tests
- Did not _fix_ anything here, just removed failing tests

rlippmann commented:
Maybe a requirement that poetry has isn't in requirements.txt?

EVWorth commented:
@HeroesDieYoung 

Try rebasing... I refactored to use uv and the workflows are updated. 

Tests are still failing but I'm curious to see if the same issues occur?

HeroesDieYoung commented:
> @HeroesDieYoung
> 
> Try rebasing... I refactored to use uv and the workflows are updated.
> 
> Tests are still failing but I'm curious to see if the same issues occur?

Updated, tests still seem to be failing the same way.

EVWorth commented:
thanks @HeroesDieYoung!

a couple of questions and thoughts:

1) what is the devcontainer.json used for?

2) I noticed you still have the requirements.txt files, these can be removed now that we're using uv and pyproject.toml no?

3) I noticed you downgraded some package versions, bumped some others, and installed multiple dev dependencies into the main dependency group.

can you speak to those dependency changes, uv remove [name] the dev packages you added to the main group and re-add them using uv add --dev [name]?
specific questions:

 > - "aiohttp>=3.9.1", - why was the minimum version "downgraded" from 3.12.14?
 > - "lxml>=5.1.0", - why was the minimum version "downgraded" from 6.0.0?
 
 > - "aiohttp-zlib-ng>=0.1.1", - should be removed. replaced by aiohttp-fast-zlib[zlib-ng]
 
  > - "pytest-isolate>=0.0.12", - should be removed and added --dev
 
 > - "aioresponses>=0.7.3", - is already be in dev group, should be removed (uv remove [name])
 > - "freezegun>=1.5.3", - is already be in dev group, should be removed (uv remove [name])
 > - "pytest-asyncio>=1.0.0", - is already be in dev group, should be removed (uv remove [name])
 > - "pytest-mock>=3.14.1", - is already be in dev group, should be removed (uv remove [name])
 > - "typeguard>=4.1.5", - is already be in dev group, should be removed (uv remove [name])

HeroesDieYoung commented:
> So I think** in your devcontainer env, before you removed it ( 😃 ), I noticed you were running uv venv, uv run requirements.txt
> 
> all of that can be replaced completely with uv sync
> 
> uv sync will build a venv with the python version specified in pyproject.toml, and then install all dependencies configured in the uv.lock file. I think** that should work in your devcontainer as well. try replacing that line with just uv sync and let me know what happens (just curious)
> 
> Otherwise, commits look good and I see passing tests! woot! thanks for the contribution!

Updated as you said, and bingo it does indeed behave exactly as you described. TIL a _tiny_ bit about `uv`. Thanks!

EVWorth commented:
@HeroesDieYoung also, I feel like I'm in the same boat. no idea what I'm doing and taking a lot of research to get things working. slowly learning more every day.. to two 😃
